backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java

```diff
@@ -710,52 +683,45 @@ public UserInfo getUserByLoginEmail(String loginEmail) {
         throw new RestrictedAccessUserException();
       }
 
-      return consolidateUser(
-          apiUser,
-          oktaUser.getOrganizationRoleClaims(),
-          oktaUser.getStatus(),
-          oktaUser.isSiteAdmin());
+      return consolidateUser(apiUser, oktaUser);
     } catch (IllegalGraphqlArgumentException | UnidentifiedUserException e) {
       throw new OktaAccountUserException();
     }
   }
 
-  private UserInfo consolidateUser(
-      ApiUser apiUser,
-      Optional<OrganizationRoleClaims> optClaims,
-      UserStatus userStatus,
-      boolean isSiteAdmin) {
-
-    OrganizationRoleClaims claims = optClaims.orElseThrow(UnidentifiedUserException::new);
-
-    // use the target user's org so response is built correctly even if site admin is the requester
-    Organization org = _orgService.getOrganization(claims.getOrganizationExternalId());
-
-    List<Facility> facilities = _orgService.getFacilities(org);
-    Set<Facility> facilitiesSet = new HashSet<>(facilities);
-
-    boolean allFacilityAccess = claims.grantsAllFacilityAccess();
-    Set<Facility> accessibleFacilities =
-        allFacilityAccess
-            ? facilitiesSet
-            : facilities.stream()
-                .filter(f -> claims.getFacilities().contains(f.getInternalId()))
-                .collect(Collectors.toSet());
-
-    OrganizationRoles orgRoles =
-        new OrganizationRoles(org, accessibleFacilities, claims.getGrantedRoles());
-    if (!_featureFlagsConfig.isOktaMigrationEnabled() && !isSiteAdmin) {
-      try {
-        setRolesAndFacilities(orgRoles, apiUser);
-      } catch (PrivilegeUpdateFacilityAccessException e) {
-        log.warn(
-            "Could not migrate roles and facilities for user with id={} because facilities were invalid",
-            apiUser.getInternalId());
+  private UserInfo consolidateUser(ApiUser apiUser, PartialOktaUser oktaUser) {
+    boolean isSiteAdmin = oktaUser.isSiteAdmin();
+    UserStatus userStatus = oktaUser.getStatus();
+    OrganizationRoleClaims oktaClaims =
+        oktaUser.getOrganizationRoleClaims().orElseThrow(UnidentifiedUserException::new);
+    OrganizationRoles orgRoles = _orgService.getOrganizationRoles(oktaClaims);
+
+    if (!isSiteAdmin) {
+      if (_featureFlagsConfig.isOktaMigrationEnabled()) {
+        orgRoles = getOrgRolesFromDB(apiUser);
+      } else {
+        setRolesAndFacilities(Optional.ofNullable(orgRoles), apiUser);
       }
     }
+
     return new UserInfo(apiUser, Optional.of(orgRoles), isSiteAdmin, userStatus);
   }
 
+  private UserInfo consolidateNonSiteAdminUser(
```

Comment:
Instead of creating an extra method here, I think we might have an opportunity to just overload the existing `consolidateUser` to take in a site admin user param and pull the function innards into a delegate method. That way we can just have the one interface for "grab all claims and store them appropriately` like so:

```
  private OrganizationRoles getOrgRoles(Optional<OrganizationRoleClaims> oktaClaims, ApiUser apiUser) {
        OrganizationRoles orgRoles = null;
        if (oktaClaims.isPresent()) {
            orgRoles = _orgService.getOrganizationRoles(oktaClaims.get());
        }
        if (_featureFlagsConfig.isOktaMigrationEnabled()) {
            orgRoles = getOrgRolesFromDB(apiUser);
        } else {
            setRolesAndFacilities(Optional.ofNullable(orgRoles), apiUser);
        }
        return orgRoles;
    }

    private UserInfo consolidateUser(
            ApiUser apiUser, Optional<OrganizationRoleClaims> oktaClaims, Boolean isSiteAdmin) {
        OrganizationRoles orgRoles = parseOrgRoles(oktaClaims, apiUser);
        return new UserInfo(apiUser, Optional.of(orgRoles), isSiteAdmin);
    }

    private UserInfo consolidateUser(
            ApiUser apiUser, Optional<OrganizationRoleClaims> oktaClaims) {
        OrganizationRoles orgRoles = parseOrgRoles(oktaClaims, apiUser);
        return new UserInfo(apiUser, Optional.of(orgRoles), false);
    }
```

And in the case where we have the existing params for `PartialOktaUser` like on 644, pass in 
`return consolidateUser(apiUser, oktaUser.getOrganizationRoleClaims(), oktaUser.isSiteAdmin())` as needed